src/platform_impl/android/mod.rs

```diff
@@ -44,6 +44,20 @@ fn poll(poll: Poll) -> Option<EventSource> {
 pub struct EventLoop<T: 'static> {
     window_target: event_loop::EventLoopWindowTarget<T>,
     user_queue: Arc<Mutex<VecDeque<T>>>,
+    first_event: Option<EventSource>,
+    start_cause: event::StartCause,
+    looper: ThreadLooper,
+    running: bool,
+}
+
+macro_rules! call_event_handler {
+    ( $event_handler:expr, $window_target:expr, $cf:expr, $event:expr ) => {{
+        if $cf != ControlFlow::Exit {
+            $event_handler($event, $window_target, &mut $cf);
+        } else {
+            $event_handler($event, $window_target, &mut ControlFlow::Exit);
+        }
+    }};
```

Comment:
Can you explain what this macro is for? The if else block seems redundant

Reply:
Its pretty much copied from the windows implementation.

if the user sets the control flow at `MainEventsCleared` but later changes the control flow to `wait` in the `RedrawRequested` event it won't exit anymore. This only allows you to change the control flow if the control flow is not exit.

This happens for example every single event with the `window_run_return` example which always sets the `*control_flow = ControlFlow::Wait;` back to wait even if the control flow is exit already. See here: 
https://github.com/rust-windowing/winit/blob/master/examples/window_run_return.rs#L31

Reply:
Oh I see, you're passing a mutable reference, my bad